Frameworks/SafariServices/SFSafariWebViewController.mm

```diff
@@ -49,13 +70,24 @@ - (BOOL)webView:(UIWebView*)webView
     shouldStartLoadWithRequest:(NSURLRequest*)request
                 navigationType:(UIWebViewNavigationType)navigationType {
     NSURL* url = request.URL;
+    BOOL openWithAppDelegate = NO;
 
     // Check whether this request is for the redirect URL that we're watching out for
     // (modulo the query string and fragment, which are allowed to vary)
-
     if ([url.scheme isEqualToString:_redirectUrl.scheme] && [url.host isEqualToString:_redirectUrl.host] &&
         [url.path isEqualToString:_redirectUrl.path]) {
-        // Let the client know we received the redirect URL
+        openWithAppDelegate = YES;
+    }
+
+    // Check whether the URL can be opened directly by this application, in which case we'll send it
+    // straight to the application delegate.
+    // Asking the WebView to open such a URL would work, since it would send an activation request
+    // to this app, but that would cause an extraneous confirmation dialog from Windows.
+    if (appCanOpenURL(url)) {
```

Comment:
Thanks for the clarification.
I'm fine leaving as is or changing it to skip the appCanOpenURL call.  
either way is fine.

---
In reply to: [83745451](https://github.com/Microsoft/WinObjC/pull/1147#discussion_r83745451) [](ancestors = 83745451,83742017)